public/app/features/datasources/components/ButtonRow.tsx

```diff
@@ -14,7 +14,7 @@ export interface Props {
   onTest: (event: any) => void;
 }
 
-const ButtonRow: FC<Props> = ({ canSave, canDelete, onDelete, onSubmit, onTest, exploreUrl }) => {
+export function ButtonRow({ canSave, canDelete, onDelete, onSubmit, onTest, exploreUrl }: Props): React.ReactElement {
```

Comment:
Genuine question... does the return type help with these components?  I though the implicit flavor ends up having more useful (and less verbose) types.

```suggestion
export function ButtonRow({ canSave, canDelete, onDelete, onSubmit, onTest, exploreUrl }: Props) {
```

Reply:
Yeah, I cannot really think of any benefit of "being explicit" here either, I think TypeScript will just infer the return types fine. (The only scenario I could imagine is that it can prevent you from accidentally returning with something you don't want to).

The reason why I have added it to the new components was to be in sync with our [FE style guide](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#how-to-declare-functional-components), so we are a bit more coherent. What do you think?

Reply:
Fixed in https://github.com/grafana/grafana/pull/51874/commits/f7dd31dc51804fd0a5df1a894ca07a01569ad42f. 👍